### DISCUSSION TITLE
Emergency Alert When Elon Musk is Near You

### DISCUSSION THREAD

**Discussion by tigercoding56** [Jxck-S/plane-notify#122] [Ideas]
just a idea i had

**Comment by Cfomodz** [comment]
Just run with your ZIP code

```bash
python proximity_alert.py 93061
```

What it does:

- Converts your ZIP code to GPS coordinates automatically
- Tracks a specific aircraft by ICAO code (currently set to Elon's jet: A835AF)
- Checks every 30 seconds using the free OpenSky Network API
- Alerts you when the aircraft comes within 80km (50 miles)
- Shows real-time distance, altitude, speed, and heading
- Prevents alert spam with a 5-minute cooldown
- Provides a direct link to track on ADSBExchange


Edit these variables at the top of the script:

    ICAO_CODE - Change to track any aircraft
    PROXIMITY_RADIUS_KM - Adjust alert distance
    CHECK_INTERVAL_SECONDS - How often to check
    ALERT_COOLDOWN_SECONDS - Time between alerts

Only uses requests - install with:

```bash
pip install requests
```

[proximity_alert.py](https://github.com/user-attachments/files/23843495/proximity_alert.py)
